### ISSUE TITLE
stdint.h inclusion issue on macOS

### ISSUE THREAD

**Issue by slimsag** [ziglang/zig#9455]
`test.c`:

```c
#include <stddef.h>
#include <stdint.h>

int main() {
   return 0;
}
```

`zig cc -lc test.c`:

```
In file included from test.c:2:
/Users/slimsag/Desktop/zig/build/lib/zig/include/stdint.h:20:43: fatal error: cannot open file '/Users/slimsag/Desktop/zig/build/lib/zig/libc/include/aarch64-macos-gnu/stdint.h': No such file or directory
#if __STDC_HOSTED__ && __has_include_next(<stdint.h>)
                                          ^
1 error generated.
```

* stage1
* f9798108f (nightly)
* macOS 11.4 / Big Sur, aarch64/m1

**Comment by kubkon** [comment]
Hmm, I'm having trouble replicating this locally on my M1 mac or x86_64 linux. Running:

```
zig cc -lc test.c
```

on macOS 11.5 produces correct executable, and similarly, running:

```
zig cc -lc test.c -target aarch64-macos
```

on x86_64 linux works fine too. Just to clarify, if you linked against brew's LLVM, did you build your Zig with either statically linked LLVM, or dynamically linked `libclang` but not both? That is, appended either `-DZIG_STATIC_LLVM=on` **or** `-DZIG_PREFER_CLANG_CPP_DYLIB=on`?

**Comment by slimsag** [comment]
thanks for taking a look @kubkon !

I suspect the difference to be that I do not have xcode installed (I have the XCode Command Line developer tools, as brew requires that, but not otherwise.)

As for building Zig, I started with a completely fresh M1 system (so no chance I have anything conflicting installed) and used:

```
brew install llvm
```

Then:

```
mkdir build
cd build
cmake .. -DCMAKE_PREFIX_PATH=$(brew --prefix llvm)
make install
```

**Comment by kubkon** [comment]
Right, so when linking against brew's LLVM you need to force either static linking of LLVM with `-DZIG_STATIC_LLVM=on` **or** dynamic linking of `libclang` with `-DZIG_PREFER_CLANG_CPP_DYLIB=on`. Otherwise, our CMake config will pick up what `llvm-config` tells it to and we end mixing `libLLVM.dylib` with `libclang.a` which causes the issue you just filed.

**Comment by kubkon** [comment]
I should add that this should have nothing to do with the presence of XCode.

**Comment by kubkon** [comment]
tl;dr, try rebuilding your Zig with the following CMake config

```
cmake .. -DCMAKE_PREFIX_PATH=$(brew --prefix llvm) -DZIG_STATIC_LLVM=on
```

**Comment by slimsag** [comment]
❗❗❗ this explains so much and it also fixes the earlier issue I was running into (#9447) I can't believe I didn't consider this sooner.

I do get a warning still, but it seems harmless:

```
 % zig cc -lc test.c                                   
warning(link): directory not found for '-L/usr/local/lib'
```

Thank you so much @kubkon ! I really appreciate the help here - I had spent upwards of 16h so far trying to debug this 😅 so if I can ever buy you a coffee sometime or something please let me know..

**Comment by kubkon** [comment]
No probs @slimsag and apologies for not making this requirement more explicit. I've now remedied this in our Wiki for building stage1 from source [here](https://github.com/ziglang/zig/wiki/Building-Zig-From-Source/_compare/bd98582ef45885a311f74e0b9b0cc9456ec7ce73...0685ac08bd927d22764a5e22c50bef8efb2faa50) and [here](https://github.com/ziglang/zig/wiki/Troubleshooting-Build-Issues/_compare/1184260dffbedac79457f8719490ab01bbef2ae8...f96fc9e36825a2c7794cc7cea9a600f8fee77323). Please have a look and let me know if this works!

Also, the warning

```
warning(link): directory not found for '-L/usr/local/lib'
```

stems from the fact that Apple removed this dir in BigSur but we search both `/usr/lib` and `/usr/local/lib` when compiling natively on macOS.

**Comment by slimsag** [comment]
That looks perfect and would've definitely worked for me! Thanks a ton again :)
